lib/translator/inquiry.cc

```diff
@@ -25,6 +25,83 @@ namespace translator {
 
 // command specific helpers
 namespace {
+
+constexpr uint8_t kIdentifierLengthNguid = 0x10;
+constexpr uint8_t kIdentifierLengthEui64 = 0x8;
+
+// Builds the IdentificationDescriptor according to 6.1.4.5 of the translation
+// spec:
+// https://nvmexpress.org/wp-content/uploads/NVM_Express_-_SCSI_Translation_Reference-1_5_20150624_Gold.pdf
+// This only builds the constant part of the Identification Descriptor. The
+// caller is responsible for building the identifier.
+StatusCode BuildIdentificationDescriptor(
+    const nvme::IdentifyNamespace& identify_namespace_data,
+    Span<uint8_t> buffer, int index, uint8_t& identifier_length) {
```

Comment:
This function seems to only write to `buffer` starting at `index`. Would it be simpler for the caller to pass in `buffer.subspan(index)` from the start?

Reply:
that's a great suggestion. Definitely cleaner. Updated the code